src/ui/App.tsx

```diff
@@ -27,31 +27,23 @@ const App = () => {
 
   const items = filterItemsSelector(store);
 
-  useEffect(() => {
-    if (downPressed || (ctrlPressed && nPressed)) {
-      dispatch({ type: "NEXT" });
-    }
+  if (downPressed || (ctrlPressed && nPressed)) {
+    dispatch({ type: "NEXT" });
```

Comment:
I was trying to avoid using `dispatch`, outside `useEffect`, but is it fine performance wise and react wise? 

But since the `downPress` and `(ctrlPressed && nPressed) will be `true` while the key is pressed, it will be hard to handle using `useEffect`...

Reply:
Regarding performance, I've not seen any negative impact.

However, I've just discovered that I've unintentionally disabled `ctrl+n` & `ctrl+p `.

Reply:
ok thanks!

Reply:
@cdes
I tried to handle `ctrl+n`, `ctrl+p`, but I couldn't find a good solution to work.

So, I took a different approach.
Creating hooks for key holds action. `useKeyHoldAction` (and `use2KeysHoldAction`).

This approach works without adding any changes to `useKeyPress.ts`. 

```typescript
const useHoldAction = (keys: boolean[], action: () => void) => {
  const [intervalId, setIntervalId] = useState<number | null>(null);
  const [timeoutId, setTimeoutId] = useState<number | null>(null);
  useEffect(() => {
    if (keys.reduce((key, acc) => key && acc, true)) {
      action();
      // Sleep a bit before dispatching action every 50 ms
      const timeoutId = setTimeout(() => {
        const intervalId = setInterval(() => {
          action();
        }, 50);
        setIntervalId(intervalId);
      }, 250);

      setTimeoutId(timeoutId);
    } else {
      clearTimeout(timeoutId);
      clearInterval(intervalId);
    }
  }, keys);
};

const useKeyHoldAction = (targetKey: string, action: () => void) => {
  const keyPressed = useKeyPress(targetKey);
  useHoldAction([keyPressed], action);
};

const use2KeysHoldAction = (
  targetKey: string,
  targetKey2: string,
  action: () => void
) => {
  const keyPressed = useKeyPress(targetKey);
  const keyPressed2 = useKeyPress(targetKey2);
  useHoldAction([keyPressed, keyPressed2], action);
};

const App = () => {
  const [store, dispatch] = useStoreReducer();
  //..
  useKeyHoldAction("ArrowDown", () => dispatch({ type: "NEXT" }));
  useKeyHoldAction("ArrowUp", () => dispatch({ type: "PREV" }));
  use2KeysHoldAction("Control", "n", () => dispatch({ type: "NEXT" }));
  use2KeysHoldAction("Control", "p", () => dispatch({ type: "PREV" }));
  // ..
}
```

Reply:
![output](https://user-images.githubusercontent.com/7858158/69840718-d9b77380-129f-11ea-8f11-4b01a50da169.gif)

Reply:
Brilliant!

However, I don’t think we should control the repeat, because it’s configurable by the OS.

Take a look at this package, it already handles repeat:
https://use-key-state.mihaicernusca.com/

I could not get it to work (maybe because of preact... I really don’t know).

Reply:
Maybe because of preact... 
I really want to change it to react.

thanks for library! I'll check the code, and see if i can make it available for preact!